tasks/src/main/scala/dagr/tasks/ScatterGather.scala

```diff
@@ -53,9 +54,41 @@ object ScatterGather {
     def sources: Option[Seq[Source]]
   }
 
+  /** Wraps a set of results as a [[Partitioner]] that produces those results as partitions. */
+  private class FixedPartitioner[Result](results: Iterable[Result]) extends SimpleInJvmTask with Partitioner[Result] {
+    var partitions: Option[Seq[Result]] = None
+    override def run(): Unit = this.partitions = Some(results.toSeq)
+  }
+
+  /** Wraps a single [[Task]] as a [[Partitioner]] that produces a single partition (the task). */
+  private class SingletonPartitioner[Result <: Task](result: Result) extends Partitioner[Result] {
+    val partitions: Option[Seq[Result]] = Some(Seq(result))
+    override def getTasks: Iterable[_ <: Task] = Some(result)
+  }
+
   /** Provides a factory method to create a [[Scatter]] object from a [[Partitioner]]. */
   object Scatter {
     def apply[Result](partitioner: Partitioner[Result]): Scatter[Result] = partitioner.scatter
+
+    /** Marker trait for [[Scatter.empty]]. */
+    private[tasks] sealed trait EmptyScatter[Result] extends Scatter[Result]
```

Comment:
Could you put the methods in here instead of in an anonymous subclass below.  Also, why is this a sealed trait and not just a `final class`.?